test/index.html

```diff
@@ -0,0 +1,247 @@
+        // However, other QUnit.test() blocks than this will fail miserably if it
+        // weren't supported.
+        assert.ok( !document.body.classList.contains("impress-disabled"),
+                   "body.impress-disabled is removed." );
+        assert.ok( document.body.classList.contains("impress-enabled"),
+                   "body.impress-enabled is added." );
+        
+        var canvas = document.querySelector( "div#impress > div" );
+        assert.ok( !canvas.classList.contains("step") && canvas.id === "",
+                   "Additional 'canvas' div inserted between div#impress root and steps." );
+        assert.equal( canvas.style.transform,
+                      "rotateZ(0deg) rotateY(0deg) rotateX(0deg) translate3d(1000px, 0px, 0px)",
+                      "canvas.style.transform initialized correctly" );
+        assert.equal( canvas.style.transformOrigin,
+                      "left top 0px",
+                      "canvas.style.transformOrigin initialized correctly" );
+        assert.equal( canvas.style.transformStyle,
+                      "preserve-3d",
+                      "canvas.style.transformStyle initialized correctly" );
+        assert.equal( canvas.style.transitionDelay,
+                      "0ms",
+                      "canvas.style.transitionDelay initialized correctly" );
+        // impress.js default values tries to set this to 1000ms, I'm completely confused about why that's not actually set in the browser?
+        assert.equal( canvas.style.transitionDuration,
+                      "0ms",
+                      "canvas.style.transitionDuration initialized correctly" );
+        assert.equal( canvas.style.transitionProperty,
+                      "all",
+                      "canvas.style.transitionProperty initialized correctly" );
+        assert.equal( canvas.style.transitionTimingFunction,
+                      "ease-in-out",
+                      "canvas.style.transitionTimingFunction initialized correctly" );
+                      
+        assert.equal( document.documentElement.style.height,
+                      "100%",
+                      "documentElement.style.height is 100%" );
+        
+        // Steps initialization
+        var step1 = document.querySelector( "div#step-1" );
+        assert.equal( step1.style.position,
+                      "absolute",
+                      "Step position is 'absolute'." );
+
+        assert.ok( step1.classList.contains("active"),
+                   "Step 1 has active css class." );
+      }
+    });
+
+    // Note: Here we focus on testing the core functionality of moving between
+    // steps, the css classes set and unset, and events triggered.
+    // TODO: more complex animations and check position, transitions, delays, etc...
+    // Those need to be separate html files, and there could be several of them.
+    QUnit.test( "Impress Core API", function( assert ) {
```

Comment:
You mentioned QUnit.module() before, and I looked into it, but at this point I don't see that it would add any value. Maybe if there are more tests for the Core API in the future, then yes. But like you now understand, that might not actually happen.

I did briefly consider an iframe, but stayed with this approach now for simplicity. Also, I've been trying to keep as close as possible to how impress.js is typically used in real world, which is to own the complete window. Probably running in an iframe would be an equivalent test, but out of principle I want to test the "real thing" as closely as possible. I do think we should keep the iframe option in mind though, especially when more tests are added.